### PR TITLE
fix(pkg/driver): do not call FixupKernel when building drivers.

### LINKED ISSUE #372: falco-driver-loader docker image fails to build on Flatcar Container Linux
**What happened**:
While building a new release of Flatcar Container Linux, we realised our tests for `falco-driver-loader` have been failing and the script has been replaced with `falcoctl`. We use the Docker image `falcosecurity/falco-driver-loader` to test same. 

It seems that the deprecated script `falco-driver-loader` had the logic using `/lib/modules/<kernel_version>-flatcar/` whereas the new command `falcoctl` seems to look for `/lib/modules/<flatcar_version>-flatcar/`.

```
core@localhost ~ $ docker run --rm --privileged -v /root/.falco:/root/.falco -v /proc:/host/proc:ro -v /boot:/host/boot:ro -v /lib/modules:/host/lib/modules:ro -v /usr:/host/usr:ro -v /etc:/host/etc:ro falcosecurity/falco-driver-loader:master                      
* Setting up /usr/src links from host
2023-12-12 11:27:12 INFO  Running falcoctl driver install
                      ├ driver version: 000d576ef877cb115cbb56f97187a1d62221e2bd
                      ├ driver type: kmod
                      ├ driver name: falco
                      ├ compile: true
                      ├ download: true
                      ├ arch: x86_64
                      ├ kernel release: 6.1.62-flatcar
                      └ kernel version: #1 SMP PREEMPT_DYNAMIC Mon Nov 20 22:57:52 -00 2023
2023-12-12 11:27:12 INFO  Found distro target: flatcar
2023-12-12 11:27:12 INFO  Check if kernel module is still loaded. 
2023-12-12 11:27:12 INFO  OK! There is no module loaded. 
2023-12-12 11:27:12 INFO  Check all versions of kernel module in dkms. 
2023-12-12 11:27:12 INFO  OK! There are no module versions in dkms. 
2023-12-12 11:27:12 INFO  Trying to download a driver.
                      └ url: https://download.falco.org/driver/000d576ef877cb115cbb56f97187a1d62221e2bd/x86_64/falco_flatcar_3794.0.0-flatcar_#1 SMP PREEMPT_DYNAMIC Mon Nov 20 22:57:52 -00 2023.ko
2023-12-12 11:27:14 WARN  Non-200 response from url. code: 404
2023-12-12 11:27:14 WARN  unable to find a prebuilt driver 
2023-12-12 11:27:14 INFO  Flatcar detected; relocating kernel tools.
                      └ version: 3794.0.0
2023-12-12 11:27:14 INFO  Trying to dkms install module. gcc: /usr/bin/gcc
Sign command: /lib/modules/3794.0.0-flatcar/build/scripts/sign-file
Binary /lib/modules/3794.0.0-flatcar/build/scripts/sign-file not found, modules won't be signed
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
Creating symlink /var/lib/dkms/falco/000d576ef877cb115cbb56f97187a1d62221e2bd/source -> /usr/src/falco-000d576ef877cb115cbb56f97187a1d62221e2bd
Error! Your kernel headers for kernel 3794.0.0-flatcar cannot be found at /lib/modules/3794.0.0-flatcar/build or /lib/modules/3794.0.0-flatcar/source.
Please install the linux-headers-3794.0.0-flatcar package or use the --kernelsourcedir option to tell DKMS where it's located.
2023-12-12 11:27:15 WARN  Running dkms build failed, couldn't find dkms log
                      └ file: /var/lib/dkms/$falco/000d576ef877cb115cbb56f97187a1d62221e2bd/build/make.log
2023-12-12 11:27:15 INFO  Trying to dkms install module. gcc: /usr/bin/gcc-11
Sign command: /lib/modules/3794.0.0-flatcar/build/scripts/sign-file
Binary /lib/modules/3794.0.0-flatcar/build/scripts/sign-file not found, modules won't be signed
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
Error! Your kernel headers for kernel 3794.0.0-flatcar cannot be found at /lib/modules/3794.0.0-flatcar/build or /lib/modules/3794.0.0-flatcar/source.
Please install the linux-headers-3794.0.0-flatcar package or use the --kernelsourcedir option to tell DKMS where it's located.
2023-12-12 11:27:15 WARN  Running dkms build failed, couldn't find dkms log
                      └ file: /var/lib/dkms/$falco/000d576ef877cb115cbb56f97187a1d62221e2bd/build/make.log
2023-12-12 11:27:15 INFO  Trying to dkms install module. gcc: /usr/bin/gcc-12
Sign command: /lib/modules/3794.0.0-flatcar/build/scripts/sign-file
Binary /lib/modules/3794.0.0-flatcar/build/scripts/sign-file not found, modules won't be signed
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
Error! Your kernel headers for kernel 3794.0.0-flatcar cannot be found at /lib/modules/3794.0.0-flatcar/build or /lib/modules/3794.0.0-flatcar/source.
Please install the linux-headers-3794.0.0-flatcar package or use the --kernelsourcedir option to tell DKMS where it's located.
2023-12-12 11:27:15 WARN  Running dkms build failed, couldn't find dkms log
                      └ file: /var/lib/dkms/$falco/000d576ef877cb115cbb56f97187a1d62221e2bd/build/make.log
2023-12-12 11:27:15 INFO  Trying to load a pre existent system module, if present.
2023-12-12 11:27:15 WARN  Consider compiling your own driver and loading it or getting in touch with the Falco community.
2023-12-12 11:27:15 ERROR failed: failed to compile the module 
``` 

The same Flatcar image works well with an older release.

```
core@localhost ~ $ docker run --rm --privileged -v /root/.falco:/root/.falco -v /proc:/host/proc:ro -v /boot:/host/boot:ro -v /lib/modules:/host/lib/modules:ro -v /usr:/host/usr:ro -v /etc:/host/etc:ro falcosecurity/falco-driver-loader:0.36.2
* Setting up /usr/src links from host
* Running falco-driver-loader for: falco version=0.36.2, driver version=6.0.1+driver, arch=x86_64, kernel release=3794.0.0, kernel version=1
* Running falco-driver-loader with: driver=module, compile=yes, download=yes

================ Cleaning phase ================

* 1. Check if kernel module 'falco' is still loaded:
- OK! There is no 'falco' module loaded.

* 2. Check all versions of kernel module 'falco' in dkms:
- OK! There are no 'falco' module versions in dkms.

[SUCCESS] Cleaning phase correctly terminated.

================ Cleaning phase ================

* Looking for a falco module locally (kernel 3794.0.0)
* Filename 'falco_flatcar_3794.0.0_1.ko' is composed of:
 - driver name: falco
 - target identifier: flatcar
 - kernel release: 3794.0.0
 - kernel version: 1
* Trying to download a prebuilt falco module from https://download.falco.org/driver/6.0.1%2Bdriver/x86_64/falco_flatcar_3794.0.0_1.ko
curl: (22) The requested URL returned error: 404
Unable to find a prebuilt falco module
* Flatcar detected (version 3794.0.0); relocating kernel tools
** Found host dl interpreter: /host/usr/lib64/ld-linux-x86-64.so.2
** Setting host dl interpreter for /lib/modules/6.1.62-flatcar/build/scripts/basic/fixdep
/usr/bin/falco-driver-loader: line 256: patchelf: command not found
** Setting host dl interpreter for /lib/modules/6.1.62-flatcar/build/scripts/mod/modpost
/usr/bin/falco-driver-loader: line 256: patchelf: command not found
** Setting host dl interpreter for /lib/modules/6.1.62-flatcar/build/tools/objtool/objtool
/usr/bin/falco-driver-loader: line 256: patchelf: command not found
install: /usr/lib/gcc/x86_64-linux-gnu/12/
* Trying to dkms install falco module with GCC /usr/bin/gcc
Sign command: /lib/modules/6.1.62-flatcar/build/scripts/sign-file
Signing key: /var/lib/dkms/mok.key
Public certificate (MOK): /var/lib/dkms/mok.pub
Certificate or key are missing, generating self signed certificate for MOK...
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
Creating symlink /var/lib/dkms/falco/6.0.1+driver/source -> /usr/src/falco-6.0.1+driver

Building module:
Cleaning build area...
'/tmp/falco-dkms-make'.....
Signing module /var/lib/dkms/falco/6.0.1+driver/build/falco.ko
Cleaning build area...

falco.ko.xz:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/6.1.62-flatcar/updates/dkms/
depmod...
* falco module installed in dkms
* falco module found: /var/lib/dkms/falco/6.0.1+driver/6.1.62-flatcar/x86_64/module/falco.ko.xz
* Trying to insmod
* Success: falco module found and loaded in dkms
```

**What you expected to happen**:

The paths are proper, and the build happens successfully.

**How to reproduce it (as minimally and precisely as possible)**:

- Grab a Flatcar QEMU image.
- Run the container image.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area library

**What this PR does / why we need it**:

Moreover, call `return f.generic.FixupKernel(kr)` in flatcar FixupKernel.

**Which issue(s) this PR fixes**:

Fixes #372 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/driver/distro/distro.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func Build(ctx context.Context,
 	if err != nil {
 		return "", err
 	}
-	path, err := driverType.Build(ctx, printer, d.FixupKernel(kr), driverName, driverVer, env)
+	path, err := driverType.Build(ctx, printer, kr, driverName, driverVer, env)
```
Comment (FedeDP) [inline pkg/driver/distro/distro.go:181]: Fixing up the kenrel is only useful when downloading.

File: pkg/driver/distro/flatcar.go [hunk 2]
```diff
@@ -93,6 +93,9 @@ func (f *flatcar) customizeBuild(ctx context.Context,
 		return nil, nil
 	}
 	printer.Logger.Info("Flatcar detected; relocating kernel tools.", printer.Logger.Args("version", f.versionID))
-	_, err := exec.CommandContext(ctx, "/bin/bash", "-c", flatcarRelocateScript).Output()
+	out, err := exec.CommandContext(ctx, "/bin/bash", "-c", flatcarRelocateScript).Output()
+	if err != nil {
+		printer.DefaultText.Print(string(out))
```
Comment (FedeDP) [inline pkg/driver/distro/flatcar.go:98]: Print the error in case this step fails.

**Review by leogr — APPROVED** [review]

**Pull request merged**